### PR TITLE
fix(domain): remove deprecated api call

### PR DESCRIPTION
**Pull request by lizardK**
| Question         | Answer
| ---------------- | ---
| Branch?          | `release/2021-w8` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #DTRSD-31971
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch 
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally 
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits 

## Description

ref: DTRSD-31971

Signed-off-by: Cyril Biencourt <cyril.biencourt@ovhcloud.com>

### CODE REVIEW

File: packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.routing.js [hunk 1]
```diff
@@ -52,11 +52,7 @@ export default /* @ngInject */ ($stateProvider) => {
         return promise;
       },
 
-      isZoneValid: /* @ngInject */ (newDnsZone) => (name) =>
-        newDnsZone
-          .getZoneNameValidation(name)
-          .then(() => true)
-          .catch(() => false),
+      isZoneValid: /* @ngInject */ () => Promise.resolve(true),
```
Comment (marie-j) [inline packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.routing.js:55]: Always ?

Comment (lizardK) [reply packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.routing.js:55]: always

Comment (cbourgois) [reply packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.routing.js:55]: As `isZoneValid` is only used in component's binding (https://github.com/ovh/manager/blob/master/packages/manager/apps/web/client/app/dns-zone/new/new.component.js#L8)
maybe we could remove `checkZoneName` & `zoneNameisValid` statements in controller and template (https://github.com/ovh/manager/blob/master/packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.controller.js#L49 & https://github.com/ovh/manager/blob/master/packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.html#L44)

**Review by cbourgois — APPROVED** [review]

File: packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.routing.js (hunk 1 above)
Comment (antleblanc) [reply packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.routing.js:55]: Good point @cbourgois!

Will take this patch as is in order to create a new release but we will apply your suggestion via a separate Pull Request.
Thanks!

**Review by antleblanc — APPROVED** [review]

**Pull request merged**